qiskit/circuit/library/generalized_gates/diagonal.py

```diff
@@ -71,21 +71,22 @@ class Diagonal(QuantumCircuit):
     `arXiv:0406176 <https://arxiv.org/pdf/quant-ph/0406176.pdf>`_
     """
 
-    def __init__(self, diag: list[complex] | np.ndarray) -> None:
+    def __init__(self, diag_dict: dict) -> None:
         """Create a new Diagonal circuit.
 
         Args:
-            diag: list of the 2^k diagonal entries (for a diagonal gate on k qubits).
+            diag: dict of the diagonal entries which are not 1.
 
         Raises:
-            CircuitError: if the list of the diagonal entries or the qubit list is in bad format;
-                if the number of diagonal entries is not 2^k, where k denotes the number of qubits
+            CircuitError: if the dictionary of the diagonal entries are in bad format;
+                          if absolute value of any entry is not 1.
         """
-        if not isinstance(diag, (list, np.ndarray)):
-            raise CircuitError("Diagonal entries must be in a list or numpy array.")
+        diag = _dict_to_list(diag_dict)
+        if not isinstance(diag_dict, dict):
+            raise CircuitError("Diagonal entries must be dictionary.")
         num_qubits = np.log2(len(diag))
-        if num_qubits < 1 or not num_qubits.is_integer():
-            raise CircuitError("The number of diagonal entries is not a positive power of 2.")
+        # if num_qubits < 1 or not num_qubits.is_integer():
+        # raise CircuitError("The number of diagonal entries is not a positive power of 2.")
```

Comment:
Why is this code commented out?

Reply:
As of new implementation `num_qubits` will always $\geq$ 1 hence this check is not valid. Forgot to remove, will resolve it.